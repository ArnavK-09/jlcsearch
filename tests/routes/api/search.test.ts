import { test, expect } from "bun:test"
import { getTestServer } from "tests/fixtures/get-test-server"

test("GET /components/list with search query 'C1234' returns expected components", async () => {
  const { axios } = await getTestServer()
  const res = await axios.get("/api/search?limit=1&q=C1234")

  expect(res.data).toHaveProperty("components")
  expect(Array.isArray(res.data.components)).toBe(true)

  expect(res.data.components).toMatchSnapshot()
})

test("GET /components/list with search query '555 Timer' returns expected components", async () => {
  const { axios } = await getTestServer()
  const res = await axios.get("/api/search?limit=1&q=555%20Timer")

  expect(res.data).toHaveProperty("components")
  expect(Array.isArray(res.data.components)).toBe(true)

  expect(res.data.components).toMatchSnapshot()
})

test("GET /components/list with search query 'red led' returns expected components", async () => {
  const { axios } = await getTestServer()
  const res = await axios.get("/api/search?limit=1&q=red led")

  expect(res.data).toHaveProperty("components")
  expect(Array.isArray(res.data.components)).toBe(true)

  expect(res.data.components).toMatchSnapshot()
})
