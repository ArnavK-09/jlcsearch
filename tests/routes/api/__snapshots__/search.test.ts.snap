// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /components/list with search query 'C1234' returns expected components 1`] = `
[
  {
    "description": "125mW Thin Film Resistor 100V ±25ppm/℃ ±1% 1.23MΩ 0805 Chip Resistor - Surface Mount ROHS",
    "lcsc": 217796,
    "mfr": "ARG05FTC1234N",
    "package": "0805",
    "price": 0.005642857,
    "stock": 9,
  },
]
`;

exports[`GET /components/list with search query '555 Timer' returns expected components 1`] = `
[
  {
    "description": "DIP-8 555 Timers / Counters ROHS",
    "lcsc": 22461592,
    "mfr": "LM555CN",
    "package": "DIP-8",
    "price": 0.141,
    "stock": 2519,
  },
]
`;

exports[`GET /components/list with search query 'red led' returns expected components 1`] = `
[
  {
    "description": "-40℃~+85℃ Red 0603 LED Indication - Discrete ROHS",
    "lcsc": 2286,
    "mfr": "KT-0603R",
    "package": "0603",
    "price": 0.005314286,
    "stock": 3208103,
  },
]
`;
